lib/http.ex

```diff
@@ -209,4 +233,48 @@ defmodule Braintree.HTTP do
       @sandbox_endpoint
     )
   end
+
+  defp emit_start(method, path) do
+    if telemetry_started() do
+      :telemetry.execute(
+        [:braintree, :request, :start],
+        %{system_time: System.system_time()},
+        %{method: method, path: path}
+      )
+    end
+  end
+
+  defp emit_exception(duration, method, path, error_data) do
+    if telemetry_started() do
+      :telemetry.execute(
+        [:braintree, :request, :exception],
+        %{duration: duration},
+        %{method: method, path: path, error: error_data}
+      )
+    end
+  end
+
+  defp emit_error(duration, method, path, error_reason) do
+    if telemetry_started() do
+      :telemetry.execute(
+        [:braintree, :request, :error],
+        %{duration: duration},
+        %{method: method, path: path, error: error_reason}
+      )
+    end
+  end
+
+  defp emit_stop(duration, method, path, code) do
+    if telemetry_started() do
+      :telemetry.execute(
+        [:braintree, :request, :stop],
+        %{duration: duration},
+        %{method: method, path: path, http_status: code}
+      )
+    end
+  end
+
+  defp telemetry_started do
```

Comment:
Telemetry is getting ubiquitous enough and is light weight enough that I think we can have it as a non-optional dependency.